nessie/sql_templates/edl_create_advisor_schema.template.sql

```diff
@@ -0,0 +1,135 @@
+ * "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ * ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+--------------------------------------------------------------------
+-- CREATE EXTERNAL SCHEMA
+--------------------------------------------------------------------
+
+CREATE EXTERNAL SCHEMA {redshift_schema_advisor}
+FROM data catalog
+DATABASE '{redshift_schema_advisor}'
+IAM_ROLE '{redshift_iam_role}'
+CREATE EXTERNAL DATABASE IF NOT EXISTS;
+
+--------------------------------------------------------------------
+-- External Tables
+--------------------------------------------------------------------
+
+-- Advisor note permissions (BOA_ADV_NOTES_ACCESS_VW)
+CREATE EXTERNAL TABLE {redshift_schema_advisor}.advisor_note_permissions
+(
+  USER_ID VARCHAR,
+  CS_ID VARCHAR,
+  PERMISSION_LIST VARCHAR
+)
+ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.OpenCSVSerde'
+WITH SERDEPROPERTIES (
+  'separatorChar' = ',',
+  'quoteChar' = '\"',
+  'escapeChar' = '\\'
+)
+STORED AS TEXTFILE
+LOCATION '{advisor_data_path}/advisor-note-permissions';
+
+--------------------------------------------------------------------
+-- Internal Schema
+--------------------------------------------------------------------
+
+DROP SCHEMA IF EXISTS {redshift_schema_advisor_internal} CASCADE;
+CREATE SCHEMA {redshift_schema_advisor_internal};
+GRANT USAGE ON SCHEMA {redshift_schema_advisor_internal} TO GROUP {redshift_app_boa_user}_group;
+ALTER default PRIVILEGES IN SCHEMA {redshift_schema_advisor_internal} GRANT SELECT ON TABLES TO GROUP {redshift_app_boa_user}_group;
+GRANT USAGE ON SCHEMA {redshift_schema_advisor_internal} TO GROUP {redshift_dblink_group};
+ALTER DEFAULT PRIVILEGES IN SCHEMA {redshift_schema_advisor_internal} GRANT SELECT ON TABLES TO GROUP {redshift_dblink_group};
+
+--------------------------------------------------------------------
+-- Internal Tables
+--------------------------------------------------------------------
+
+CREATE TABLE {redshift_schema_advisor_internal}.advisor_departments
+SORTKEY (sid)
+AS (
+    SELECT
+        advisor_id AS sid,
+        advisor_role AS advisor_type_code,
+        advisor_role_desc AS advisor_type,
+        academic_plan_cd AS plan_code,
+        academic_plan_nm AS plan,
+        academic_department_cd AS department_code,
+        academic_department_short_nm AS department
+    FROM {redshift_schema_edl_external}.student_advisor_data
+    WHERE
+        academic_career_cd='UGRD'
+        AND advisor_id ~ '[0-9]+'
+        AND academic_department_cd IS NOT NULL
+        AND academic_plan_cd IS NOT NULL
+    ORDER BY advisor_id, academic_plan_cd
+);
+
+CREATE TABLE {redshift_schema_advisor_internal}.advisor_roles
+SORTKEY (sid)
+AS (
+    SELECT DISTINCT
+        p.CS_ID AS sid,
+        p.USER_ID AS uid,
+        a.advisor_role AS advisor_type_code,
+        a.advisor_role_desc AS advisor_type,
+        NULL AS instructor_type_code,
+        NULL AS instructor_type,
```

Comment:
`instructor_type` is not available in EDL but is it necessary? I kept NULLs above as placeholders. The values are used by neither Nessie nor BOA.